kustomize/commands/internal/util/util_test.go

```diff
@@ -95,6 +95,47 @@ func TestGlobPatternsWithLoaderRemoteFile(t *testing.T) {
 	}
 }
 
+func TestIsHiddenFilePath(t *testing.T) {
+	hiddenFilePaths := []string{
+		".*",
+		"/.*",
+		"dir/.*",
+	}
+	filePaths := []string{
+		"*",
+		"/*",
+		"dir/*",
+	}
+	for _, path := range hiddenFilePaths {
+		res := IsHiddenFilePath(path)
+		if !res {
+			t.Fatalf("Hidden file path not correctly identified: %s\n", path)
+		}
+	}
+	for _, path := range filePaths {
+		res := IsHiddenFilePath(path)
+		if res {
+			t.Fatalf("Normal file path incorrectly identified as hidden file path: %s\n", path)
+		}
+	}
+}
+
+func TestRemoveHiddenFiles(t *testing.T) {
+	paths := []string{
+		"dir/fa1",
+		"dir/fa2",
+		"dir/.fa3",
```

Comment:
Suggestion: for completeness include top-level files (hidden & not) and files with extensions (hidden & not) to show those cases are handled correctly too.